### PR TITLE
Default args for facades should be `js.native`

### LINKED ISSUE #562: All facade default arguments should be `js.native`
Enforce via scalafix

### PR DESCRIPTION
**Pull request by japgolly**
Closes #562

### CODE REVIEW

File: scalafix/src/main/scala/org/scalajs/dom/scalafix/GenerateApiReport.scala [hunk 1]
```diff
@@ -95,7 +95,7 @@ class GenerateApiReport extends SemanticRule("GenerateApiReport") {
           .replaceAll(" {2,}", " ")
           .trim
           .stripSuffix(" = js.native")
-          .replaceAll(" = js.native(?=[^\n])", "?")
+          .replaceAll(" = js\\.(native|undefined)(?=[^\n])", "?")
```
Comment (armanbilge) [inline scalafix/src/main/scala/org/scalajs/dom/scalafix/GenerateApiReport.scala:98]: Nice! this was an eyesore for me too :)

File: src/main/scala/org/scalajs/dom/OffscreenCanvas.scala [hunk 2]
```diff
@@ -16,7 +16,7 @@ class OffscreenCanvas(var width: Double, var height: Double) extends js.Object {
   def getContext(contextType: String, contextAttributes: TwoDContextAttributes): js.Dynamic = js.native
 
   /** Creates a Blob object representing the image contained in the canvas. */
-  def convertToBlob(options: ConvertToBlobOptions = ???): js.Promise[Blob] = js.native
+  def convertToBlob(options: ConvertToBlobOptions = js.native): js.Promise[Blob] = js.native
```
Comment (armanbilge) [inline src/main/scala/org/scalajs/dom/OffscreenCanvas.scala:19]: Crazy!!

**Review by armanbilge — APPROVED** [review]
Yep this is a big improvement overall. Thanks for moving fast on this!

File: src/main/scala/org/scalajs/dom/OffscreenCanvas.scala (hunk 2 above)
Comment (japgolly) [reply src/main/scala/org/scalajs/dom/OffscreenCanvas.scala:19]: Haha yeah ikr!

**Comment by armanbilge** [comment]
@japgolly btw how are you working around https://github.com/scala-js/scala-js/issues/4553?

File: scalafix/src/main/scala/org/scalajs/dom/scalafix/DefaultFacadeArgs.scala [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  override def fix(implicit doc: SemanticDocument): Patch =
+    doc.tree.collect {
+
+      case Defn.Def(mods, _, _, paramss, _, body) if Util.isJsNative(body) =>
+
+        // https://github.com/scala-js/scala-js/issues/4553
+        // `js.native` as default arg doesn't compile for top-level method facades.
+        if (Util.isJSGlobal(mods)) {
+          // Ignore for now
+          Patch.empty
```
Comment (armanbilge) [inline scalafix/src/main/scala/org/scalajs/dom/scalafix/DefaultFacadeArgs.scala:17]: Ah, I see. Hmm.

Comment (japgolly) [reply scalafix/src/main/scala/org/scalajs/dom/scalafix/DefaultFacadeArgs.scala:17]: Ah yeah, was just about to link that. Not sure what else we can do for now. I assume it's just an accidental oversight and @sjrd will fix it for the next release. My plan is when that happens we can remove this condition.

Comment (armanbilge) [reply scalafix/src/main/scala/org/scalajs/dom/scalafix/DefaultFacadeArgs.scala:17]: Seems good to me! We'll continue to keep an eye out for this in reviews in the meantime.

**Pull request merged**